devtools/cli/src/main/java/io/quarkus/cli/registry/RegistryListCommand.java

```diff
@@ -0,0 +1,37 @@
+package io.quarkus.cli.registry;
+
+import io.quarkus.devtools.project.QuarkusProjectHelper;
+import io.quarkus.registry.config.RegistriesConfig;
+import io.quarkus.registry.config.RegistriesConfigLocator;
+import io.quarkus.registry.config.RegistryConfig;
+import picocli.CommandLine;
+
+@CommandLine.Command(name = "list", sortOptions = false, showDefaultValues = true, mixinStandardHelpOptions = false, header = "List enabled Quarkus registries", description = "%n"
+        + "This command will list currently enabled Quarkus extension registries", headerHeading = "%n", commandListHeading = "%nCommands:%n", synopsisHeading = "%nUsage: ", parameterListHeading = "%n", optionListHeading = "Options:%n")
+public class RegistryListCommand extends BaseRegistryCommand {
+
+    @CommandLine.Option(names = { "--refresh" }, description = "Refresh the local Quarkus extension registry cache")
+    boolean refresh;
+
+    @Override
+    public Integer call() throws Exception {
+        if (refresh) {
+            output.debug("Refreshing registry cache");
+            try {
+                output.throwIfUnmatchedArguments(spec.commandLine());
+                QuarkusProjectHelper.setMessageWriter(output);
+                QuarkusProjectHelper.getCatalogResolver().clearRegistryCache();
+            } catch (Exception e) {
+                return output.handleCommandException(e,
+                        "Unable to refresh the registry cache: " + e.getMessage());
+            }
+        }
+
+        output.info("Available Quarkus extension registries:");
+        final RegistriesConfig config = RegistriesConfigLocator.resolveConfig();
```

Comment:
Should the BaseRegistryCommand also bring in the RegistryClient mixin? And shouldn't that mixin be used to find the catalog resolver so that it is consistent?

Reply:
The `registry` command isn't actually using the registry client. At least at this point. It's basically a config reading/editing command.

Reply:
`QuarkusProjectHelper.getCatalogResolver().clearRegistryCache();` -- that is what the RegistryClient currently encapsulates..

Reply:
Perhaps, once we enable the registry client by default that could make more sense. While all of this is related to the registry client, the `RegistryClientMixin` is currently used to provide the `--registry-client` option and `QuarkusProject` with an `ExtensionCatalog`. Atm, none of that is relevant for the registry command.

Reply:
hmm. true. but would/should `--refresh` apply everywhere that the registry client is used?

Reply:
Ok, refactored to add the `--refresh` option to the RegistryClientMixin.